docs/DesignDocs/SGX_QuoteVerify_Integration.md

```diff
@@ -0,0 +1,488 @@
+    size_t quote_size,
+    oe_datetime_t expiration_check_date,
+    [out] uint32_t *p_collateral_expiration_status,
+    [out] uint32_t *p_quote_verification_result,
+    [in, out, size=qve_report_size] void* p_qve_report_info,
+    size_t qve_report_size,
+    [out, size=supplemental_data_size] void* p_supplemental_data,
+    size_t supplemental_data_size,
+    [out] size_t* p_supplemental_data_size_out);
+```
+
+### Update implementation of existing plugin functions
+
+These plugin functions include `_verify_evidence()`, `_verify_report()`
+and `oe_verify_quote_with_sgx_endorsements()`.
+
+- Update API `oe_verify_quote_with_sgx_endorsements()` implementation as below:
+  - Input parameter doesn't change.
+  - In host-side plugin:
+    - Set input parameter `p_qve_report_info` to NULL, then call host-side
+      function `oe_sgx_verify_quote()`, which will call SGX QVL library to
+      verify quote.
+  - In enclave-side plugin:
+    - Construct structure `sgx_ql_qe_report_info_t` as below, `nonce` and
+      `app_enclave_target_info` are input, qve_report is QvE report which
+      target to application enclave.
+    ```C
+    typedef struct _sgx_ql_qe_report_info_t {
+      sgx_quote_nonce_t nonce;
+      sgx_target_info_t app_enclave_target_info;
+      sgx_report_t qve_report;
+    }sgx_ql_qe_report_info_t;
+    ```
+    - Call OCALL function `oe_verify_quote_ocall()` to host-side plugin library
+      to verify SGX quote.
+    - With all the QvE return values, call new API.
+      `oe_verify_qve_report_and_identity()` (described below) to
+      verify QvE report and identity.
+- In API `_verify_evidence()`, remove function call `oe_get_sgx_endorsements()`,
+  and only call `oe_parse_sgx_endorsements()` if caller provides an endorsement buffer
+- In API `_verify_report()` and relevant internal APIs, such as
+  `oe_verify_report_internal()` and `oe_verify_sgx_quote()`, remove the code that
+  loads QPL and retrieves endorsements.
+
+### Add function in enclave-side plugin for QvE Identity verification
+
+For enclave-side verifier plugin, after quote verification, it needs to verify that the result
+comes from a trusted QvE.
+
+Intel® SGX SDK provides a library named `sgx_dcap_tvl` to help the verifier to
+verify QvE's identity, you can refer to [source file](https://github.com/intel/SGXDataCenterAttestationPrimitives/blob/master/QuoteVerification/dcap_tvl/sgx_dcap_tvl.cpp)
+
+This library uses hardcoded QvE identity values, because:
+- There is no need for x509 and JSON parser in verifier enclave's TCB.
+- Most of QvE's identity attributes only change infrequently.
+
+The only identity info may change more often is QvE's ISV SVN, so the verifier needs
+to provide a SVN number as threshold, only when current QvE's ISV SVN is equal to
+or larger than this threshold, the verifier can trust the QvE verification result.
+The verifier can get latest QvE ISV SVN from [QvE Identity at Intel® Provisioning
```

Comment:
This seems to be a direct parallel to the QEIdentity, which we explicitly treat as part of the collateral bundle handled during attestation. Is there a reason why this isn't treated the same way?